src/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -187,12 +186,12 @@ internal static X509Store EnsureStoreOpened(bool isMachineStore)
         {
             X509Store store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
 
-            if (store == null)
+            if (Volatile.Read(ref store) == null)
             {
                 lock (s_syncObject)
                 {
                     store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
-                    if (store == null)
+                    if (Volatile.Read(ref store) == null)
```

Comment:
That note refers to fields with the `volatile` specifier, which is overly strong in this case (as the second read doesn't need its own memory barrier). Before `Volatile.{Read,Write}`, however, there was no other way to perform volatile reads/writes.

Regardless, it's worth investigating whether or not Lazy<T> is a better choice.